Language server crash when initializing on windows

After building regal.exe on MS windows from commit f566a4e8, I get errors when opening my workspace in vs code.

```text
[Error - 1:36:49 PM] Client Regal LSP client: connection to server is erroring.
write EPIPE
[Error - 1:36:49 PM] Server initialization failed.
  Message: write EPIPE
  Code: -32099 
[Error - 1:36:49 PM] Regal LSP client client: couldn't create connection to server.
  Message: write EPIPE
  Code: -32099 
Regal Language Server (path: c:\git\work\policies\build\regal.exe, version: Unknown)
panic: failed to parse ref: expected exactly one term but got:  [recovered, repanicked]

goroutine 1 [running]:
github.com/open-policy-agent/regal/internal/lsp/rego.init.OnceValue[...].func2.1.1()
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/oncefunc.go:63 +0x75
panic({0x7ff70d1eb080?, 0xc0005212c0?})
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/runtime/panic.go:783 +0x132
github.com/open-policy-agent/regal/internal/compile.init.OnceValue[...].func2.1.1()
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/oncefunc.go:63 +0x75
panic({0x7ff70d1eb080?, 0xc0005212c0?})
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/runtime/panic.go:783 +0x132
github.com/open-policy-agent/opa/v1/ast.MustParseRef(...)
	C:/Users/geirs/go/pkg/mod/github.com/open-policy-agent/opa@v1.8.0/v1/ast/parser_ext.go:119
github.com/open-policy-agent/regal/internal/compile.init.func1.1({0xc00051bca0, 0x1c}, 0xc000c1bbd8)
	C:/git/others/regal/internal/compile/compile.go:54 +0x1c5
github.com/open-policy-agent/regal/internal/io/files.(*WalkReducer[...]).ReduceFS.func1()
	C:/git/others/regal/internal/io/files/walker.go:198 +0x2c
github.com/open-policy-agent/regal/internal/io/files.(*Walker).WalkFS.(*Walker).walker.func1({0xc00051bca0, 0x1c}, {0x7ff70dc869f0, 0x7ff70dca34d8}, {0x0?, 0x0?})
	C:/git/others/regal/internal/io/files/walker.go:132 +0x1c9
io/fs.walkDir({0x7ff70dc778e0, 0x7ff70dca3460}, {0xc00051bca0, 0x1c}, {0x7ff70dc869f0, 0x7ff70dca34d8}, 0xc000c87008)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/io/fs/walk.go:73 +0x6c
io/fs.walkDir({0x7ff70dc778e0, 0x7ff70dca3460}, {0xc000b183a0, 0xd}, {0x7ff70dc869f0, 0x7ff70dca34a8}, 0xc000c87008)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/io/fs/walk.go:95 +0x2bc
io/fs.walkDir({0x7ff70dc778e0, 0x7ff70dca3460}, {0x7ff70d3a797f, 0x7}, {0x7ff70dc85e30, 0xc00099c0b0}, 0xc000c87008)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/io/fs/walk.go:95 +0x2bc
io/fs.WalkDir({0x7ff70dc778e0, 0x7ff70dca3460}, {0x7ff70d3a797f, 0x7}, 0xc000c87008)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/io/fs/walk.go:122 +0x9a
github.com/open-policy-agent/regal/internal/io/files.(*Walker).WalkFS(0xc00090ed80, {0x7ff70dc778e0, 0x7ff70dca3460}, 0xc000c87058)
	C:/git/others/regal/internal/io/files/walker.go:99 +0x6d
github.com/open-policy-agent/regal/internal/io/files.(*WalkReducer[...]).ReduceFS(0x0?, {0x7ff70dc778e0?, 0x7ff70dca3460?}, 0x7ff70d1e15e0?)
	C:/git/others/regal/internal/io/files/walker.go:197 +0x54
github.com/open-policy-agent/regal/internal/compile.init.func1()
	C:/git/others/regal/internal/compile/compile.go:48 +0x345
github.com/open-policy-agent/regal/internal/compile.init.OnceValue[...].func2.1()
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/oncefunc.go:66 +0x58
sync.(*Once).doSlow(0x0?, 0x0?)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/once.go:78 +0xac
sync.(*Once).Do(0x0?, 0x0?)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/once.go:69 +0x19
github.com/open-policy-agent/regal/internal/compile.init.OnceValue[...].func2()
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/oncefunc.go:58 +0x45
github.com/open-policy-agent/regal/internal/lsp/rego.init.func1()
	C:/git/others/regal/internal/lsp/rego/rego.go:404 +0x29
github.com/open-policy-agent/regal/internal/lsp/rego.init.OnceValue[...].func2.1()
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/oncefunc.go:66 +0x58
sync.(*Once).doSlow(0x7ff70c3fcba5?, 0x18?)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/once.go:78 +0xac
sync.(*Once).Do(0xc000c87588?, 0xc000c87588?)
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/once.go:69 +0x19
github.com/open-policy-agent/regal/internal/lsp/rego.init.OnceValue[...].func2()
	C:/Users/geirs/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.0.windows-amd64/src/sync/oncefunc.go:58 +0x45
github.com/open-policy-agent/regal/internal/lsp/rego.SchemaResolvers(...)
	C:/git/others/regal/internal/lsp/rego/rego.go:139
github.com/open-policy-agent/regal/internal/lsp/rego.prepareQueryArgs({0x7ff70dc85ea0, 0xc000142000}, {0xc00010e078, 0x1, 0x1}, {0x7ff70dc90420, 0xc0001420a0}, 0xc00014e3c0)
	C:/git/others/regal/internal/lsp/rego/rego.go:311 +0x342
github.com/open-policy-agent/regal/internal/lsp/rego.prepareQuery({0x7ff70dc85ea0, 0xc000142000}, {0xc00010e078, 0x1, 0x1}, {0x7ff70dc90420, 0xc0001420a0})
	C:/git/others/regal/internal/lsp/rego/rego.go:356 +0x97
github.com/open-policy-agent/regal/internal/lsp/rego.StoreCachedQuery({0x7ff70dc85ea0, 0xc000142000}, {0x7ff70d3cf029, 0x1f}, {0x7ff70dc90420, 0xc0001420a0})
	C:/git/others/regal/internal/lsp/rego/rego.go:265 +0x92
github.com/open-policy-agent/regal/internal/lsp/completions/providers.NewPolicy({0x7ff70dc85ea0?, 0xc000142000?}, {0x7ff70dc90420?, 0xc0001420a0?})
	C:/git/others/regal/internal/lsp/completions/providers/policy.go:27 +0x3d
github.com/open-policy-agent/regal/internal/lsp/completions.NewDefaultManager({0x7ff70dc85ea0, 0xc000142000}, 0xc000142050, {0x7ff70dc90420, 0xc0001420a0})
	C:/git/others/regal/internal/lsp/completions/manager.go:37 +0x1b0
github.com/open-policy-agent/regal/internal/lsp.NewLanguageServerMinimal({0x7ff70dc85ea0, 0xc000142000}, 0xc000c87ad0, 0x0)
	C:/git/others/regal/internal/lsp/server.go:177 +0x145
github.com/open-policy-agent/regal/internal/lsp.NewLanguageServer({0x7ff70dc85ea0?, 0xc000142000?}, 0x7ff70d3e7309?)
	C:/git/others/regal/internal/lsp/server.go:154 +0x25
github.com/open-policy-agent/regal/cmd.init.3.func1({0xc000280900?, 0x7ff70d3a5010?, 0x0?})
	C:/git/others/regal/cmd/languageserver.go:51 +0x298
github.com/open-policy-agent/regal/cmd.init.3.wrapProfiling.func2(0xc000220c08, {0x0, 0x0, 0x0})
	C:/git/others/regal/cmd/profiling.go:16 +0x6fe
github.com/spf13/cobra.(*Command).execute(0xc000220c08, {0x7ff70e8361c0, 0x0, 0x0})
	C:/Users/geirs/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0xb02
github.com/spf13/cobra.(*Command).ExecuteC(0x7ff70e7d3200)
	C:/Users/geirs/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(0x7ff700000000?)
	C:/Users/geirs/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071 +0x13
main.main()
	C:/git/others/regal/main.go:18 +0x2c
[Error - 1:36:49 PM] Server process exited with code 2.
[Error - 1:36:49 PM] Connection to server got closed. Server will not be restarted.
```

--- a/README.md
+++ b/README.md
@@ -15,6 +15,12 @@
 
 <!-- markdownlint-disable MD041 -->
 
+Regal is a linter, debugger and language server for [Rego](https://www.openpolicyagent.org/docs/policy-language/),
+making your Rego magnificent, and you the ruler of rules!
+
+With its extensive set of linter rules, documentation and editor integrations, Regal is the perfect companion for policy
+development, whether you're an experienced Rego developer or just starting out.
+
 > [!IMPORTANT]
 > Regal is now an official OPA project!
 > ([More Info](https://blog.openpolicyagent.org/note-from-teemu-tim-and-torin-to-the-open-policy-agent-community-2dbbfe494371))
@@ -23,12 +29,6 @@
 > Regal's documentation to openpolicyagent.org from the next release. Please
 > [file an issue](https://github.com/open-policy-agent/regal/issues/new)
 > for any problems you run into.
-
-Regal is a linter, debugger and language server for [Rego](https://www.openpolicyagent.org/docs/policy-language/),
-making your Rego magnificent, and you the ruler of rules!
-
-With its extensive set of linter rules, documentation and editor integrations, Regal is the perfect companion for policy
-development, whether you're an experienced Rego developer or just starting out.
 
 
 <!-- markdownlint-disable MD033 -->

--- a/docs/readme-sections/intro.md
+++ b/docs/readme-sections/intro.md
@@ -1,5 +1,11 @@
 <!-- markdownlint-disable MD041 -->
 
+Regal is a linter, debugger and language server for [Rego](https://www.openpolicyagent.org/docs/policy-language/),
+making your Rego magnificent, and you the ruler of rules!
+
+With its extensive set of linter rules, documentation and editor integrations, Regal is the perfect companion for policy
+development, whether you're an experienced Rego developer or just starting out.
+
 > [!IMPORTANT]
 > Regal is now an official OPA project!
 > ([More Info](https://blog.openpolicyagent.org/note-from-teemu-tim-and-torin-to-the-open-policy-agent-community-2dbbfe494371))
@@ -8,9 +14,3 @@
 > Regal's documentation to openpolicyagent.org from the next release. Please
 > [file an issue](https://github.com/open-policy-agent/regal/issues/new)
 > for any problems you run into.
-
-Regal is a linter, debugger and language server for [Rego](https://www.openpolicyagent.org/docs/policy-language/),
-making your Rego magnificent, and you the ruler of rules!
-
-With its extensive set of linter rules, documentation and editor integrations, Regal is the perfect companion for policy
-development, whether you're an experienced Rego developer or just starting out.

--- a/internal/compile/compile.go
+++ b/internal/compile/compile.go
@@ -1,7 +1,6 @@
 package compile
 
 import (
-	"os"
 	"strings"
 	"sync"
 
@@ -50,7 +49,11 @@ var RegalSchemaSet = sync.OnceValue(func() *ast.SchemaSet {
 
 			util.Must0(encoding.JSON().Unmarshal(util.Must(embeds.SchemasFS.ReadFile(path)), &schemaAny))
 
-			spl := strings.Split(strings.TrimSuffix(path, ".json"), string(os.PathSeparator))
+			// > This is unlike io/fs.WalkDir, which always uses slash separated paths.
+			// https://pkg.go.dev/path/filepath#WalkDir
+			//
+			// https://github.com/open-policy-agent/regal/issues/1679
+			spl := strings.Split(strings.TrimSuffix(path, ".json"), "/")
 			ref := ast.Ref([]*ast.Term{ast.SchemaRootDocument}).Extend(ast.MustParseRef(strings.Join(spl[1:], ".")))
 
 			schemaSet.Put(ref, schemaAny)

--- a/internal/io/files/walker.go
+++ b/internal/io/files/walker.go
@@ -192,6 +192,8 @@ func (fwr *WalkReducer[T]) Reduce(f func(string, T) (T, error)) (T, error) {
 // calling f for each file. The function f receives the path of the file and
 // the current value of the accumulator. It returns the new value of the
 // accumulator and an error if any.
+// NOTE: Unlike that path passed to f is always /-separated when walking an FS
+// implementation (as opposed to a directory, where the separator is OS-specific).
 func (fwr *WalkReducer[T]) ReduceFS(target fs.FS, f func(string, T) (T, error)) (T, error) {
 	curr := fwr.initial
 	err := fwr.WalkFS(target, func(path string) (err error) {

